eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java

```diff
@@ -861,6 +861,22 @@ public String getHostName() {
         return hostName;
     }
 
+    /**
+     * Return the default network address to connect to this instance.
+     *
+     * The address can either be a hostname or an ip and there is no guarantee which will be returned.
+     * Assume the address can change dynamically over time.
+     * If a usecase need more specific hostnames or ips, please use data from {@link #getDataCenterInfo()}.
+     *
+     * For legacy reasons, the data backing this field is extracted from the hostname field
+     *
+     * @return either a hostname or an ipAddress
+     */
+    @JsonIgnore
+    public String getDefaultAddress() {
```

Comment:
Ideally, it would be nice to get rid of the toplevel hostname/ipAddr fields and just have users access them from DataCenterInfo. At the toplevel, what's nice to have is a default "address" that can be either hostname or ip, as defined by the registering service itself. Unfortunately, for backwards compatibility, the hostname field is today the realistic data that's used for this. Introducing this new accessor for a cleaner client API.